src/hooks/useAdjacentElementBorderRadius.ts

```diff
@@ -0,0 +1,103 @@
+/* External dependencies */
+import {
+  useState,
+  useEffect,
+} from 'react'
+
+function flattenBorderTopRadiusStyle(elem: HTMLElement) {
+  elem.style.borderTopLeftRadius = '0px'
+  elem.style.borderTopRightRadius = '0px'
+}
+
+function flattenBorderBottomRadiusStyle(elem: HTMLElement) {
+  elem.style.borderBottomLeftRadius = '0px'
+  elem.style.borderBottomRightRadius = '0px'
+}
+
+function resetBorderTopRadiusStyle(elem: HTMLElement) {
+  elem.style.borderTopLeftRadius = ''
+  elem.style.borderTopRightRadius = ''
+}
+
+function resetBorderBottomRadiusStyle(elem: HTMLElement) {
+  elem.style.borderBottomLeftRadius = ''
+  elem.style.borderBottomRightRadius = ''
+}
+
+function useAdjacentElementBorderRadius<ElementType extends HTMLElement>(
+  element: ElementType | null,
+  filterAdjacentElementFunc: (element: ElementType) => boolean,
+  isFilteredElement?: boolean,
+) {
+  const [nextSibling, setNextSibling] = useState<ElementType | null>(null)
+  const [previousSibling, setPreviousSibling] = useState<ElementType | null>(null)
+
+  // eslint-disable-next-line prefer-arrow-callback
+  useEffect(function updateAdjacentElement() {
+    if (!element) { return }
+
+    function filterAdjacentElement(targetElement: ElementType | null) {
+      if (!targetElement) { return null }
+      return filterAdjacentElementFunc(targetElement) ? targetElement : null
+    }
+
+    setNextSibling(filterAdjacentElement(element.nextSibling as ElementType))
+    setPreviousSibling(filterAdjacentElement(element.previousSibling as ElementType))
+  }, [
+    element,
+    filterAdjacentElementFunc,
+    isFilteredElement,
+  ])
+
+  // eslint-disable-next-line prefer-arrow-callback
+  useEffect(function stylingElementBorderRadius() {
```

Comment:
동사로 써주는 게 자연스럽지 않을까 생각합니다.
```suggestion
  useEffect(function styleElementBorderRadius() {
```